documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,322 @@
++=============================+==========================================================+
+| ``bool IsAscending``        | Sort order. Ascending: ``true``; Descending: ``false``.  |
++-----------------------------+----------------------------------------------------------+
+| ``std::uint8_t RadixBits``  | Number of bits to sort per a radix sort algorithm pass.  |
+|                             | Only ``8`` is currently supported.                       |
++-----------------------------+----------------------------------------------------------+
+
+
+Parameters
+----------
+
++------------------------------------------------------+------------------------------------------------------------------+
+| Name                                                 | Description                                                      |
++======================================================+==================================================================+
+|  ``q``                                               | SYCL queue to submit the kernel template to.                     |
++------------------------------------------------------+------------------------------------------------------------------+
+|                                                      | The sequences(s) of elements to apply the algorithm to.          |
+|  - ``rng`` (1)                                       | They can be provided as:                                         |
+|  - ``first``, ``last`` (2)                           |                                                                  |
+|  - ``keys_rng``, ``vals_rng`` (3)                    | - ``sycl::buffer`` (1,3),                                        |
+|  - ``keys_first``, ``keys_last``, ``vals_first`` (4) | - ``oneapi::dpl::experimental::ranges::views::all`` (1,3)        |
+|                                                      | - ``oneapi::dpl::experimental::ranges::views::subrange`` (1,3)   |
+|                                                      | - USM pointer (2,4)                                              |
+|                                                      | - ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` (2,4)          |
++------------------------------------------------------+------------------------------------------------------------------+
+|  ``param``                                           | Kernel configuration structure. ``data_per_workitem``,           |
+|                                                      | can be any value among ``32``, ``64``, ``96``,..., ``k * 32``;   |
+|                                                      | ``workgroup_size`` can be only ``64``.                           |
++------------------------------------------------------+------------------------------------------------------------------+
+
+
+Return Value
+------------
+
+``sycl::event`` object representing a status of the algorithm execution.
+
+
+Memory Usage
```

Comment:
If the single workgroup implementation requires a different amount of memory than the multi-workgroup implementation, we should probably choose our language accordingly, "may allocate up to" or "may require up to".  

If we think it is worthwhile to be fully transparent with this info for performance sake, perhaps we could list the memory requirements for each (although, perhaps that is too much and should only be done if we expose the individual algorithms with individual KT).

Reply:
@dmitriy-sobolev how do the single workgroup implementation memory needs compare to the multi workgroup implementation?  (For local and global memory)
If the single workgroup version requires more memory for either in any situation, that certainly needs to be included.  If it is always lower requirements for both types of memory, I think we can omit it.

Reply:
Single workgroup implementation does not allocate temporary global memory. It allocates local memory, but slightly different from multiple workgroup implementation. I will document it.

Reply:
It's been documented.